### DISCUSSION TITLE
Different Quickbooks Versions

### DISCUSSION THREAD

**Discussion by remgr** [neuralpain/qbactivator#15] [Q&A]
Hello, I understand that this will work specifically for the POS Quickbooks, will it work for other versions?  I have a retail version of Quickbooks Accountant, with 6 licenses and product codes.  All the allowed installs are used.  Can I run this to install to another PC?

Thanks

**Comment by neuralpain** [comment]
Hello. I recently published a beta update for this with a general activation option. Test it and let me know when you have any issues.

**Reply by remgr** [reply]
Hello,
Thanks for your beta update.  While I had found an old validation code that we got from Intuit, that worked, I decided to give this a shot at home.  Your script worked perfectly!  

What you may want to add onto your readme instructions, that people need to go to: Help->
About, and as the logo starts up - press Ctrl, R & P at the same time to get to input the 999999 validation code in.  I had already known how to do this from before.  

Can I take this to mean that it will work for QB 2021?  We have used up the licenses for that as well.

Thanks for your awesome script!
![QB15 Activation](https://github.com/neuralpain/qbactivator/assets/114819659/414fab3d-104e-486d-8f11-e5ea2be0a8ef)

**Reply by neuralpain** [reply]
Glad to hear that it works and I'll add your suggestion to the instructions for the activator.

In regards to the other QB versions, it should work for 2021 and others as well, however, I do not have any licenses to provide for these just yet so it's a matter of trial and error for the other products to be activated.

**Reply by modem-19** [reply]
@neuralpain @Navish360   I discovered this script last week and it looks to be amazing, esp for the beta's includes of "Other" Quickbooks versions.  However, when I install any of those (QB 2016-2023) and run the script, those QB installs are crippled and unable to run.  They start with the QB splash screen, then errors out requiring a fresh reinstall.  Which works until running the latest beta script against those 'other' versions.

Any suggestions for a workaround?

**Reply by neuralpain** [reply]
Sorry, but right now I do not have any functioning solutions for the other desktop versions of QuickBooks.

**Reply by modem-19** [reply]
@neuralpain No worries.  Thanks for taking the time to respond.  I'll plan to do some tinkering with the regular Pro Desktop versions.  I've got all editions back to 2010 and will test and tinker a bit.  If I run across anything worthwhile, I'll share here for everyone.

**Comment by pearson00** [comment]
I used your script to try to activate my copy of QB Pro 2017, after my old activation code failed and support was unable to generate a new code. the script corrupts the installation and QB won't run until I reinstall or repair it. Any idea how to solve that?

**Reply by j-gow** [reply]
Did you ever have any success with this in the end?

**Comment by remgr** [comment]
I believe you have to install the program first, close it without opening
it after the install.  Start the activator script, then after it says to
use 999999 as the validation code, bring up QB. I used 2018.  In 2018 you
have to go to Help-About, and when the QB logo start hit Ctrl r and p
together.  It gives the dialogue box to input the code.  This worked on
QB2018.  I don't have the 2017 version.  Hope this helps.

**Comment by pearson00** [comment]
Wow, thanks for the quick reply. I wish i had 2018  but I only own 2017 with license codes. I followed your procedure but the script corrupts the 2017 install and I can’t run QB without repairing it which overrides the script :/

**Comment by remgr** [comment]
Hi,

I don't know what the issue may be.  I am not the author of the script.  I
had reached out to the author, to ask if it would work on a regular QB
version vs only a POS, and he created a beta script which worked for me.  I
am the IT person in my office, and if I had your problem, I would look to
do a *COMPLETE* uninstall, including the registry.  The steps are explained
here: *https://smallbusiness.chron.com/uninstall-quickbooks-unregister-modules-56743.html
<https://smallbusiness.chron.com/uninstall-quickbooks-unregister-modules-56743.html>
*.
 I would then download a new version of your software here:
*https://www.smbaccountants.com/download-quickbooks-2017/
<https://www.smbaccountants.com/download-quickbooks-2017/> .  *This is
assuming that you have the license and product number that came with your
purchase.  If not, take a look at this video
https://www.youtube.com/watch?v=HsZVLOxOUxA&t=12s&ab_channel=HectorGarciaCPA
He provides a license number and product code for the trial.  I believe a
trial version can be converted to a full license.  So maybe try using the
script with those numbers.  We had purchased 6 licenses for the program, so
maybe that's why I had success.

I have attached a screenshot.  *Do Not *use the trial validation code, only
the license and product number.  Remember to not allow the program to start
after installation...

That's all I can offer.  You should reach out directly to
"neutalpain"through github.  He is the originator.

Good luck...

**Reply by tommydog35** [reply]
I have QuickBooks Pro 2016 (UK edition) and the activator is just not working (windows 10 64bit).  When I try your option 3 (general quickbooks activation), it seems to initially patch, but the "follow the steps below to activate quickbooks software" cannot be done.  When I open Quickbooks, after the patching I get:

Microsoft visual c++ runtime library
Runtime Error!
abnormal program termination

What is causing this?  Any idea of a fix?  I am pretty desperate to get this working.

**Comment by tommydog35** [comment]
I have QuickBooks Pro 2016 (UK edition) and the activator is just not working (windows 10 64 bit).  When I try your option 3 (general quickbooks activation), it seems to initially patch, but the "follow the steps below to activate quickbooks software" cannot be done.  When I open Quickbooks, after the patching I get:

Microsoft visual c++ runtime library
Runtime Error!
abnormal program termination

What is causing this?  Any idea of a fix?  I am pretty desperate to get this working.

**Comment by remgr** [comment]
Not sure, but I would reinstall Microsoft c++ as a troubleshooting step.
That's all I can offer.

**Reply by tommydog35** [reply]
I just reinstalled Microsoft c++, but still same problem.

**Comment by remgr** [comment]
No it is for 64 bites...

**Reply by tommydog35** [reply]
any idea what that issue is?  I even ran it in a virtual machine and same problem

**Comment by remgr** [comment]
Again, I would try to remove and reinstall all versions C++.  See link
below for steps:

https://www.landfx.com/kb/computers-networking/computer/6101-reinstall-visual-c-runtime#:~:text=In%20the%20Apps%20%26%20features%20dialog,the%20Uninstall%20button%20that%20appears
.

I would also go to GitHub and see if you can reach the author of the
activator directly.  I don't know how to fix this, maybe he does.

**Reply by tommydog35** [reply]
Thanks for the suggestions.  I have tried all of these, but it does not cure the problem.

**Comment by remgr** [comment]
No, I had no luck either...

**Comment by modem-19** [comment]
@remgr Do you still have a beta copy of that activator?  I came back here to download and try it, but don't see any publicly available copy of it.

**Reply by remgr** [reply]
Here you go:


[qbactivator-0.20.2-beta1.zip](https://github.com/user-attachments/files/26649596/qbactivator-0.20.2-beta1.zip)

**Reply by modem-19** [reply]
@remgr Thanks!

**Comment by remgr** [comment]
👍

**Comment by modem-19** [comment]
@remgr  So far I've tested the script successfully with Desktop Pro 2016.  Was going through a list of QB versions to test against.  So far Desktop Pro 2017-2020 I've ran into an issue.  Which is I install QB successfully, run qbactivator and it indicates it patches successfully.  Then when I run QB 2017-2020, the splash screen starts, then disappears and never actually runs.  I did test by disabling the internet, installing QB fresh and it runs.  Then running the activator and afterwards the same behavior happens.

I was wondering if you had seen this behavior or if there are certain versions the activator doesn't work on?

**Comment by remgr** [comment]
It is my experience that it only works on QB 2016.  I had the same results
as you.  I long ago gave up on the script.

<http://www.avg.com/email-signature?utm_medium=email&utm_source=link&utm_campaign=sig-email&utm_content=webmail>
Virus-free.www.avg.com
<http://www.avg.com/email-signature?utm_medium=email&utm_source=link&utm_campaign=sig-email&utm_content=webmail>
<#DAB4FAD8-2DD7-40BB-A1B8-4E2AA1F9FDF2>
